custom/icds/location_reassignment/dumper.py

```diff
@@ -0,0 +1,102 @@
+import io
+
+from couchexport.export import export_raw
+
+from corehq.apps.locations.models import SQLLocation
+from custom.icds.location_reassignment.const import (
+    EXTRACT_OPERATION,
+    MERGE_OPERATION,
+    MOVE_OPERATION,
+    SPLIT_OPERATION,
+)
+
+
+class Dumper(object):
+    def __init__(self):
+        """
+        Dump all transitions in an excel sheet in a format easy to understand by users
+        One tab per location type, with changes specific to the location type
+        ToDo: add TestDumper
+        See TestDumper.test_dump for example
+        """
+        self.new_site_codes = []
+
+    def dump(self, transitions_per_location_type):
+        """
+        :param transitions_per_location_type: location types mapped to transitions where
+        each transition is a dict with an operation
+        like merge or split mapped to details for the operation
+        Check Parser for the expected format for each operation
+        """
+        location_types = list(transitions_per_location_type.keys())
+        headers = [[location_type, ['from', 'transition', 'to', 'Missing', 'Archived']]
+                   for location_type in location_types]
+        stream = io.BytesIO()
+        self._setup_site_codes(list(transitions_per_location_type.values()))
+        rows = [(k, v) for k, v in self._rows(transitions_per_location_type).items()]
+        export_raw(headers, rows, stream)
+        return stream
+
+    def _setup_site_codes(self, transitions):
```

Comment:
I have the same sorts of comments about this class - instance variables are being defined inside methods and referenced elsewhere, so there's an unstated order in which things must be called.

Reply:
there is only one public method which controls that order so that would keep that clear?